src/libstrongswan/plugins/botan/botan_rsa_private_key.c

```diff
@@ -397,7 +397,7 @@ static bool calculate_pq(botan_mp_t *n, botan_mp_t *e, botan_mp_t *d,
 {
 	botan_mp_t k = NULL, one = NULL, r = NULL, zero = NULL, two = NULL;
 	botan_mp_t n1 = NULL, x = NULL, y = NULL, g = NULL, rem = NULL;
-	botan_rng_t rng = NULL;
+	botan_rng_t rng;
```

Comment:
This is actually necessary because we use `goto` in this function and if there is an error before the RNG is initialized, we would crash when trying to destroy this uninitialized RNG.

Reply:
But Botan will error out with a `nullptr` `botan_rng_t` being passed, see https://github.com/randombit/botan/blob/release-2/src/lib/ffi/ffi_rng.cpp#L25. So to me it looks like the whole function is never actually executed. That is actually a thing I wanted to point out already when pushing this change earlier this week.

Reply:
No, it errors out if you pass it a NULL pointer (i.e. if you'd call `botan_rng_init(NULL, "system")`), not if the passed pointer points to NULL.

Reply:
You're right.